hummingbot/client/settings.py

```diff
@@ -29,84 +33,75 @@
 SCRIPTS_PATH = "scripts/"
 
 
-def _get_exchanges(cex: bool = True) -> Set[str]:
-    invalid_names = ["__pycache__", "paper_trade"]
-    exchanges = set()
-    package_dir = Path(__file__).resolve().parent.parent.parent
-    connectors = [f.name for f in scandir(f'{str(package_dir)}/hummingbot/connector/exchange') if
-                  f.is_dir() and f.name not in invalid_names]
-    for connector in connectors:
-        try:
-            path = f"hummingbot.connector.exchange.{connector}.{connector}_utils"
-            is_cex = getattr(importlib.import_module(path), "CENTRALIZED")
-            if cex and is_cex:
-                exchanges.add(connector)
-            elif not cex and not is_cex:
-                exchanges.add(connector)
-        except Exception:
-            continue
-    return exchanges
-
+class ConnectorType(Enum):
+    Connector = 1
+    Exchange = 2
+    Derivative = 3
 
-def _get_derivatives() -> Set[str]:
-    invalid_names = ["__pycache__"]
-    derivatives = set()
-    try:
-        package_dir = Path(__file__).resolve().parent.parent.parent
-        connectors = [f.name for f in scandir(f'{str(package_dir)}/hummingbot/connector/derivative')
-                      if f.is_dir() and f.name not in invalid_names]
-        derivatives.update(connectors)
-    except Exception:
-        pass
-    return derivatives
 
+class ConnectorSetting(NamedTuple):
+    name: str
+    type: ConnectorType
+    example_pair: str
+    centralised: bool
+    use_ethereum_wallet: bool
+    fee_type: TradeFeeType
+    fee_token: str
+    default_fees: List[Decimal]
+    config_keys: List[ConfigVar]
```

Comment:
I think this should be `config_keys: Dict[str, ConfigVar]`

Reply:
Good catch, thanks.